code/libraries/joomlatools/component/koowa/user/user.php

```diff
@@ -15,52 +15,49 @@
  */
 final class ComKoowaUser extends KUser implements ComKoowaUserInterface
 {
-    protected function _initialize(KObjectConfig $config)
-    {
-        $user = JFactory::getUser();
-
-        $config->append(array(
-            'data' => $this->_mapData($user)
-        ));
-
-        parent::_initialize($config);
-    }
+    use ComKoowaUserTrait;
 
     /**
-     * User setter
-     *
-     * @param JUser $user A joomla user object
+     * Constructor
      *
-     * @return $this
+     * @param KObjectConfig $config An optional KObjectConfig object with configuration options.
      */
-    public function setUser(JUser $user)
+    public function __construct(KObjectConfig $config)
     {
-        $this->setData($this->_mapData($user));
+        KObject::__construct($config);
 
-        return $this;
+        //Set the user properties and attributes
+        $this->setData(JFactory::getUser());
     }
 
     /**
-     * Joomla user to Koowa user data mapper
-     *
-     * @param JUser $user
+     * Set the user data
      *
-     * @return array Koowa user data
+     * @param  array|JUser $user An associative array of data or a JUser object
+     * @return $this
      */
-    protected function _mapData(JUser $user)
+    public function setData($user)
     {
-        return array(
-            'id'         => $user->id,
-            'email'      => $user->email,
-            'name'       => $user->name,
-            'username'   => $user->username,
-            'password'   => $user->password,
-            'salt'       => '',
-            'authentic'  => !$user->guest,
-            'enabled'    => !$user->block,
-            'expired'    => !$user->activation,
-            'attributes' => $user->getParameters()->toArray()
-        );
+        if($user instanceof JUser)
```

Comment:
```suggestion
        if($user instanceof JUser || $user instanceof \Joomla\CMS\User\User)
```

@johanjanssens Just to be sure

Reply:
Agreed